src/invidious/videos/parser.cr

```diff
@@ -226,17 +230,34 @@ def parse_video_info(video_id : String, player_response : Hash(String, JSON::Any
   views_txt ||= video_details["viewCount"]?.try &.as_s || ""
   views = views_txt.gsub(/\D/, "").to_i64?
 
-  length_txt = (microformat["lengthSeconds"]? || video_details["lengthSeconds"])
+  length_txt = (microformat["lengthSeconds"]? || video_details["lengthSeconds"]?)
     .try &.as_s.to_i64
 
   published = microformat["publishDate"]?
-    .try { |t| Time.parse(t.as_s, "%Y-%m-%d", Time::Location::UTC) } || Time.utc
+    .try { |t| Time.parse(t.as_s, "%Y-%m-%d", Time::Location::UTC) }
+
+  if published.nil?
+    published_txt = video_primary_renderer
+      .try &.dig?("dateText", "simpleText")
+
+    if published_txt.try &.as_s.includes?("ago") && !published_txt.nil?
+      published = decode_date(published_txt.as_s.lchop("Started streaming "))
+    elsif published_txt && published_txt.try &.as_s.matches?(/(\w{3} \d{1,2}, \d{4})$/)
+      published = Time.parse(published_txt.as_s.match!(/(\w{3} \d{1,2}, \d{4})$/)[0], "%b %-d, %Y", Time::Location::UTC)
+    else
+      published = Time.utc
+    end
```

Comment:
Here I'd recommend using `"scheduledStartTime"` (see below) in the player response, as it's more accurate and doesn't need complex parsing logic.

```json
{
	"playabilityStatus": {
		"status": "LIVE_STREAM_OFFLINE",
		"reason": "This live event will begin in 32 hours.",
		"playableInEmbed": true,
		"liveStreamability": {
			"liveStreamabilityRenderer": {
				"videoId": "hUoukmX_BRQ",
				"offlineSlate": {
					"liveStreamOfflineSlateRenderer": {
						"scheduledStartTime": "1729468800",
						"mainText": {
							"runs": [
								{"text": "Live in "},
                                {"text": "32 hours"}
							]
						},
						"subtitleText": {
							"simpleText": "October 21 at 12:00 AM"
						}
					}
				}
			}
		}
	}
}
```

Reply:
Note: if you plan to fallback to the `"subtitleText"` value, there is a non-breaking space before AM/PM:
![image](https://github.com/user-attachments/assets/64dc768c-eded-45c3-bcc1-c7af119a3beb)